packages/contracts-bedrock/src/dispute/DisputeGameFactory.sol

```diff
@@ -133,6 +133,49 @@ contract DisputeGameFactory is OwnableUpgradeable, IDisputeGameFactory, ISemver
         uuid_ = Hash.wrap(keccak256(abi.encode(_gameType, _rootClaim, _extraData)));
     }
 
+    /// @inheritdoc IDisputeGameFactory
+    function findLatestGames(
+        GameType _gameType,
+        uint256 _start,
+        uint256 _n
+    )
+        external
+        view
+        returns (GameSearchResult[] memory games_)
+    {
+        // If the `_start` index is greater than or equal to the game array length or `_n == 0`, return an empty array.
+        if (_start >= _disputeGameList.length || _n == 0) return games_;
+
+        // Allocate enough memory for the full array, but start the array's length at `0`. We may not use all of the
+        // memory allocated, but we don't know ahead of time the final size of the array.
+        assembly {
+            games_ := mload(0x40)
+            mstore(0x40, add(games_, add(0x20, shl(0x05, _n))))
+        }
+
+        // Perform a reverse linear search for the `_n` most recent games of type `_gameType`.
```

Comment:
hmm, why do this in reverse if we want games added since `_start`?

Reply:
We want the most _recent_ games, and the dispute game list is ordered by time inserted. The reverse search starts from `_start` and then looks backwards for games added at or before `_start` 👍